packages/garbo/src/tasks/dailyItems.ts

```diff
@@ -138,6 +140,62 @@ function pickCargoPocket(): void {
   }
 }
 
+function bestDevilerCandy(): Item {
+  // These are replenishable notrade nodiscard items that we would prefer to use
+  const priorityUntradeableNoDiscardList =
+    $items`sugar shotgun, sugar shillelagh, sugar shank, sugar chapeau, sugar shorts, sugar shield, sugar shirt`.filter(
+      (i) => itemAmount(i) > 1,
+    );
+  const bestPriorityCandy =
+    priorityUntradeableNoDiscardList.length > 0
+      ? maxBy(priorityUntradeableNoDiscardList, (i) => itemAmount(i))
+      : null;
+
+  if (bestPriorityCandy) return bestPriorityCandy;
+
+  const bestCandyFromMall = maxBy(
+    Item.all().filter((i) => i.candy && i.tradeable),
+    mallPrice,
+    true,
+  );
+  // These are notrade items that have an autosell value that we don't mind using if they are the cheapest
+  const safeUntradeableCandies = $items`Comet Pop, black candy heart, peanut brittle shield`;
+  // Find the best candy from inventory, accounting for value of autosell when mall min
+  const inventoryCandies = Item.all().filter((i) =>
+    i.candy && have(i) && !i.tradeable
+      ? safeUntradeableCandies.includes(i)
+      : true,
+  );
+  const sortedInventoryCandies = inventoryCandies
+    .sort((a, b) => garboValue(a) - garboValue(b))
+    .slice(0, 50);
+  // remove any candies that cost more than the lowest
+  const prunedInventoryCandies = sortedInventoryCandies.filter(
+    (i) => garboValue(i) === garboValue(sortedInventoryCandies[0]),
+  );
+  // Lowest garbovalue candy that we have the largest amount of
+  const bestInventoryCandy =
+    prunedInventoryCandies.length > 0
+      ? maxBy(prunedInventoryCandies, itemAmount)
+      : null;
+
+  return bestInventoryCandy
+    ? maxBy(
+        [bestInventoryCandy, bestCandyFromMall],
+        (i) => (i === bestInventoryCandy ? garboValue(i) : mallPrice(i)),
+        true,
+      )
+    : bestCandyFromMall;
```

Comment:
```suggestion
  const bestInventoryCandy = () => {
    if (inventoryCandies.length === 0) return null;
    const naiveBest = maxBy(inventoryCandies, garboValue, true);
    return maxBy(inventoryCandies.filter((c) => garboValue(c) === naiveBest(c)), itemAmount);
  }();
  
  return maxBy([bestInventoryCandy, bestCandyFromMall].filter(notNull), mallPrice, true);
```
I think this is smoother?

Reply:
I'm not confident there's value in slicing the array if we aren't caching the array itself
Also, the pattern I use to define `bestInventoryCandy` here is what's called a [closure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures). The only utility we gain from using a closure here is being able to use the early return instead of the ternary, so I'd be willing to not use it. But I feel like it reads nicer this way?